datarepo-clienttests/src/main/java/runner/config/TestUserSpecification.java

```diff
@@ -31,12 +32,25 @@ public static TestUserSpecification fromJSONFile(String resourceFileName) throws
     TestUserSpecification testUser =
         objectMapper.readValue(inputStream, TestUserSpecification.class);
 
+    // read in the server file
+    String delegatorSaEnvVarOverride = readDelegatorSAFileEnvironmentVariable();
+    if (delegatorSaEnvVarOverride != null) {
+      testUser.delegatorServiceAccountFile = delegatorSaEnvVarOverride;
+    }
     testUser.delegatorServiceAccount =
         ServiceAccountSpecification.fromJSONFile(testUser.delegatorServiceAccountFile);
```

Comment:
```suggestion
    // override the service account if the environment variable is defined, otherwise use the test user one
    String delegatorSaEnvVarOverride = readDelegatorSAFileEnvironmentVariable();
    if (delegatorSaEnvVarOverride != null) {
      testUser.delegatorServiceAccountFile = delegatorSaEnvVarOverride;
      testUser.delegatorServiceAccount = ServiceAccountSpecification.fromJSONFile(delegatorSaEnvVarOverride);
    } else {
      testUser.delegatorServiceAccount = ServiceAccountSpecification.fromJSONFile(testUser.delegatorServiceAccountFile);
    }
```

Reply:
This should be equivalent but I got confused between `testUser.delegatorServiceAccountFile` and `testUser.delegatorServiceAccount` since the variables look so similar. I also updated the comment.

Reply:
Made the suggested change and added a more detailed comment - let me know if it makes sense.